### PR TITLE
 Add a utility class to handle pro classes

### PR DESCRIPTION
**Pull request by n3world**
Currently there is a single method which assits in loading a pro class but that
still requires the caller to retrieve the constructor and do the construction.
The ProClass utility class wraps that and simplifies the error handling for
the common use cases.

### CODE REVIEW

File: src/frontend/org/voltdb/utils/ProClass.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+     */
+    public static final ErrorHandler HANDLER_CRASH = (m, t) -> VoltDB.crashLocalVoltDB(m, true, t);
+
+    private final Class<T> m_proClass;
+    private ErrorHandler m_errorHandler = HANDLER_CRASH;
+    private boolean m_loadConstructor = true;
+    private Constructor<T> m_constructor;
+
+    /**
+     * Try to load a PRO class. If it's running the community edition, an error message will be logged if not
```
Comment (manjujames) [inline src/frontend/org/voltdb/utils/ProClass.java]: 'if not suppressed'?

File: src/frontend/org/voltdb/utils/ProClass.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                    + feature + '.', e);
+        }
+
+        return (ProClass<T>) NO_CLASS;
+    }
+
+    /**
+     * Construct a new instance of {@code className} using {@code parameters}. If the class cannot be loaded
+     * {@code null} is returned. If the class cannot be constructed
+     * {@link VoltDB#crashLocalVoltDB(String, boolean, Throwable)} will be invoked with the instantiation exception.
```
Comment (manjujames) [inline src/frontend/org/voltdb/utils/ProClass.java]: Need to update the java doc?

File: src/frontend/org/voltdb/utils/ProClass.java [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+     * @param parameters  to pass to the constructor of {@code className}
+     * @return an instance of {@code T}
+     */
+    public static <T> T newInstanceOf(String className, String feature, ErrorHandler loadFailure,
+            Object... parameters) {
+        return ProClass.<T>load(className, feature, loadFailure).newInstance(parameters);
+    }
+
+    private ProClass(Class<T> proClass) {
+        super();
```
Comment (manjujames) [inline src/frontend/org/voltdb/utils/ProClass.java]: Nit: Do we need the super() call here?

**Comment by manjujames** [comment]
Nice work. Thanks for doing this.

File: src/frontend/org/voltdb/utils/ProClass.java (hunk 3 above)
Comment (n3world) [reply src/frontend/org/voltdb/utils/ProClass.java]: Probably not it is just the way eclipse by default auto generates constructors it always puts a call into super();

File: src/frontend/org/voltdb/utils/ProClass.java (hunk 2 above)
Comment (n3world) [reply src/frontend/org/voltdb/utils/ProClass.java]: A little but the part about crash being invoked on failure to instantiate is still true.

File: src/frontend/org/voltdb/utils/ProClass.java (hunk 1 above)
Comment (n3world) [reply src/frontend/org/voltdb/utils/ProClass.java]: Updated to say the errorHandler will be invoked.

**Pull request merged**